module/VuFind/src/VuFind/Controller/AbstractBase.php

```diff
@@ -705,8 +705,8 @@ protected function setFollowupUrlToReferer(bool $allowCurrentUrl = true, array $
             'lbreferer',
             $this->getRequest()->getServer()->get('HTTP_REFERER', null)
         );
-        // Get the referer -- if it's empty, there's nothing to store!
-        if (empty($referer)) {
+        // Get the referer -- if it's empty or invalid, there's nothing to store!
+        if (empty($referer) || !$this->isLocalUrl($referer)) {
```

Comment:
It seems that we already have validation right below. It also does a bit of normalization, it seems. I think we may want that in isLocalUrl (and get rid of the double validation.. :) ).

Reply:
Good catch, @EreMaijala -- I think I've straightened this out now.